### PR TITLE
lint public tuple structs with private fields

### PR DESCRIPTION
**Pull request by oli-obk**
idea by @nox

### CODE REVIEW

File: tests/compile-fail/auxiliary/pub_tuple_struct_with_private_fields.rs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+pub struct Foo(i32);
+
+pub fn foo() -> Foo { Foo(42) }
```
Comment (nox) [inline tests/compile-fail/auxiliary/pub_tuple_struct_with_private_fields.rs]: New line at end of file.

**Comment by oli-obk** [comment]
apparently nightly rust doesn't allow `let Foo(_) = fn_returning_tuple_struct_with_private_fields();` anymore. So this entire lint is obsolete now.

**Comment by nox** [comment]
@oli-obk Even if the tuple has some public fields?

**Comment by oli-obk** [comment]
I'll check

**Comment by oli-obk** [comment]
yea, even then it's a compiler error: http://play.integer32.com/?gist=3617a7dfe4b32814073d9548b977d516&version=nightly

**Pull request closed without merging**